backend/utils/file_utils.py

```
@@ -66,7 +71,13 @@ def save_file(file, folder, filename):
         file.save(filepath)
         return filepath
     except (OSError, IOError) as e:
-        print(f"Error saving file: {e}")
+        logger.error("Error saving file", extra=safe_extra_fields({
+            'filename': filename,
+            'upload_folder': upload_folder,
+            'error_type': type(e).__name__,
+            'error_details': str(e),
+            'operation_result': 'failed'
+        }))
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)